utility/data_fetcher.go

```diff
@@ -0,0 +1,167 @@
+package utility
+
+import (
+	"errors"
+	"time"
+)
+
+var (
+	typeErr = errors.New("Invalid type for field.")
+	nilErr  = errors.New("Mandatory field missing.")
+)
+
+type DataFetcher struct {
+	Err error
+}
+
+func (d *DataFetcher) Float64(base map[string]interface{}, key string, keys ...string) float64 {
+	val := d.getDeep(base, keys...)[key]
+	if valFloat, ok := val.(float64); ok {
+		return valFloat
+	} else if val == nil {
+		d.Err = nilErr
+	} else {
+		d.Err = typeErr
+	}
+	return 0.0
+}
+
+func (d *DataFetcher) IntPtr(base map[string]interface{}, key string, keys ...string) *int {
+	val := d.getDeep(base, keys...)[key]
+	if val == nil {
+		return nil
+	} else if valFloat, ok := val.(float64); ok {
+		valInt := int(valFloat)
+		if valFloat == float64(valInt) {
+			return &valInt
+		}
+	} else if valFloat, ok := val.(float32); ok {
+		valInt := int(valFloat)
+		if valFloat == float32(valInt) {
+			return &valInt
+		}
+	} else if valInt, ok := val.(int); ok {
+		return &valInt
+	}
+	d.Err = typeErr
+	return nil
+}
+
+func (d *DataFetcher) Interface(base map[string]interface{}, key string, keys ...string) interface{} {
+	return d.getDeep(base, keys...)[key]
+}
+
+func (d *DataFetcher) Int(base map[string]interface{}, key string, keys ...string) int {
+	if val := d.IntPtr(base, key, keys...); val != nil {
```

Comment:
this should be enough:
```return *d.IntPtr(base, key, keys...)```

Reply:
`d.IntPtr` could be `nil` here, so I cannot just return `*d.IntPtr`.

Reply:
That is fair if you want to make sure it doesn't panic, but what difference makes the 
`val := d.getDeep(base, keys...)[key]` line?
couldn't just return 0 if val is nil?

Reply:
oh you were referring to this line, that I already changed, just haven't pushed yet.